### DISCUSSION TITLE
Why does <script setup> break v-ripple?

### DISCUSSION THREAD

**Discussion by bklik** [quasarframework/quasar#13180] [General - Components / Directives / etc] [answered]
This works fine:
```
<template><div class="test" v-ripple>{{ testVar }}</div><template>
<script>
export default {
  setup() {
    const testVar = 'Test';
    return { testVar }
  }
}
</script>
```
This breaks if using the `v-ripple` directive:
```
<template><div class="test" v-ripple>{{ testVar }}</div><template>
<script setup>
  const testVar = 'Test';
</script>
```

I get the following error in the console:
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'config')
```

What am I not understanding about the Composition API and setup?

**Comment by rstoenescu** [comment, marked as answer]
It has nothing to do with script setup or Composition API.
It is a bug in Vue that has been recently introduced. Please follow our updates on https://github.com/quasarframework/quasar/issues/13154
